Maven build failed as args is outdated. Need to update source and target option to 1.7 or later

### Describe the bug
When I was following the [server deployment guide](https://www.sofastack.tech/en/projects/sofa-registry/server-quick-start/) from Sofa-registry to build the application via maven, an error was thrown

### Expected behavior

The maven should build the project successfully without any errors.

### Actual behavior

Errors were thrown below.

```bash
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project registry-core: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
```

```bash
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project registry-client-api: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
```

### Steps to reproduce
1. open command line
2. do the following commands:
```bash
git clone https://github.com/sofastack/sofa-registry.git
cd sofa-registry
mvn clean package -DskipTests
```
3. error throws

### Minimal yet complete reproducer code (or GitHub URL to code)

### Environment

- SOFARegistry version: 5.4.2 (pulled the latest commit in master)
- JVM version (e.g. `java -version`): 
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment (build 16.0.1+9-24)
OpenJDK 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
- OS version (e.g. `uname -a`): Macbook pro m1 13-inch, macOS Big Sur v11.4
- Maven version: 3.8.1
- IDE version:

### Suggested fix:
On the project that throws maven errors, update the version to 1.8 in pom.xml to fix the issue:
```
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
```
But it is not tested properly whether it will cause any further runtime errors when the registry is started.

Hi @NickNYU, I tried to use my suggested fix to build the sofa registry, but eventually find out my suggested fix cannot startup the registry service.

**Step to reproduce what I did:**
I update the following pom.xml files:
```
client/all/pom.xml
client/pom.xml
core/pom.xml
```

and run again 
`mvn clean package -DskipTests`
It was successfully built.

However, if I proceed to start the server using `sh bin/startup.sh`. The following message will be shown:

```bash
Command: java  -Dregistry.integration.home=/ProjectPath/sofa-registry-integration/registry-integration -Dspring.config.location=/ProjectPath/sofa-registry-integration/registry-integration/conf/application.properties -Duser.home=/ProjectPath/sofa-registry-integration/registry-integration -server -Xms512m -Xmx512m -Xmn256m -Xss256k -XX:+DisableExplicitGC -XX:+PrintGCDetails -XX:+PrintGCDateStamps -Xloggc:/ProjectPath/sofa-registry-integration/registry-integration/logs/registry-integration-gc.log -verbose:gc -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/ProjectPath/sofa-registry-integration/registry-integration/logs -XX:ErrorFile=/ProjectPath/sofa-registry-integration/registry-integration/logs/registry-integration-hs_err_pid%p.log -XX:-OmitStackTraceInFastThrow -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:ParallelGCThreads=4 -XX:+CMSClassUnloadingEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=70 -jar /ProjectPath/sofa-registry-integration/registry-integration/registry-integration.jar  --logging.config=/ProjectPath/sofa-registry-integration/registry-integration/conf/logback-spring.xml
Unrecognized VM option 'PrintGCDateStamps'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

Therefore I added `XX:+IgnoreUnrecognizedVMOptions` to the [startup.sh](https://github.com/sofastack/sofa-registry/blob/abd97d161724720774b40f548a050c81e7f920bb/server/distribution/session/bin/startup.sh).
```sh
mkdir -p "${BASE_DIR}/logs"
JAVA_OPTS="$JAVA_OPTS -XX:+DisableExplicitGC -XX:+PrintGCDetails -XX:+PrintGCDateStamps -Xloggc:${BASE_DIR}/logs/registry-${APP_NAME}-gc.log -verbose:gc"
JAVA_OPTS="$JAVA_OPTS -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${BASE_DIR}/logs -XX:ErrorFile=${BASE_DIR}/logs/registry-${APP_NAME}-hs_err_pid%p.log"
JAVA_OPTS="$JAVA_OPTS -XX:-OmitStackTraceInFastThrow -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:ParallelGCThreads=4"
JAVA_OPTS="$JAVA_OPTS -XX:+CMSClassUnloadingEnabled -XX:+UseCMSInitiatingOccupancyOnly -XX:CMSInitiatingOccupancyFraction=70"
JAVA_OPTS="$JAVA_OPTS -XX:+IgnoreUnrecognizedVMOptions" # added this
```

Finally, the service failed to start. The following error was thrown:

#### common-error.log
```bash
[2021-07-20 15:36:05,205][ERROR][main][SpringApplication] - Application startup failed
java.lang.IllegalStateException: Cannot load configuration class: com.alipay.sofa.registry.server.meta.MetaApplication
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.enhanceConfigurationClasses(ConfigurationClassPostProcessor.java:404)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanFactory(ConfigurationClassPostProcessor.java:249)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:283)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:127)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:687)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:525)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:134)
	at com.alipay.sofa.registry.server.integration.RegistryApplication.main(RegistryApplication.java:70)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: java.lang.ExceptionInInitializerError: null
	at org.springframework.context.annotation.ConfigurationClassEnhancer.newEnhancer(ConfigurationClassEnhancer.java:122)
	at org.springframework.context.annotation.ConfigurationClassEnhancer.enhance(ConfigurationClassEnhancer.java:110)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.enhanceConfigurationClasses(ConfigurationClassPostProcessor.java:394)
	... 18 common frames omitted
Caused by: org.springframework.cglib.core.CodeGenerationException: java.lang.reflect.InaccessibleObjectException-->Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @2471cca7
	at org.springframework.cglib.core.ReflectUtils.defineClass(ReflectUtils.java:464)
	at org.springframework.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:336)
	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:93)
	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:91)
	at org.springframework.cglib.core.internal.LoadingCache$2.call(LoadingCache.java:54)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.springframework.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:61)
	at org.springframework.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:116)
	at org.springframework.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:291)
	at org.springframework.cglib.core.KeyFactory$Generator.create(KeyFactory.java:221)
	at org.springframework.cglib.core.KeyFactory.create(KeyFactory.java:174)
	at org.springframework.cglib.core.KeyFactory.create(KeyFactory.java:153)
	at org.springframework.cglib.proxy.Enhancer.<clinit>(Enhancer.java:73)
	... 21 common frames omitted
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @2471cca7
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:357)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199)
	at java.base/java.lang.reflect.Method.setAccessible(Method.java:193)
	at org.springframework.cglib.core.ReflectUtils$1.run(ReflectUtils.java:61)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:554)
	at org.springframework.cglib.core.ReflectUtils.<clinit>(ReflectUtils.java:52)
	at org.springframework.cglib.core.KeyFactory$Generator.generateClass(KeyFactory.java:243)
	at org.springframework.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
	at org.springframework.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:329)
	... 33 common frames omitted
```

#### common-default.log
```bash
[2021-07-20 15:36:02,159][INFO][main][SpringApplication] - 
______           _     _                _____
| ___ \         (_)   | |              /  ___|
| |_/ /___  __ _ _ ___| |_ _ __ _   _  \ `--.  ___ _ ____   _____ _ __
|    // _ \/ _` | / __| __| '__| | | |  `--. \/ _ \ '__\ \ / / _ \ '__|
| |\ \  __/ (_| | \__ \ |_| |  | |_| | /\__/ /  __/ |   \ V /  __/ |
\_| \_\___|\__, |_|___/\__|_|   \__, | \____/ \___|_|    \_/ \___|_|
            __/ |                __/ |
           |___/                |___/


[2021-07-20 15:36:02,188][INFO][background-preinit][Version] - HV000001: Hibernate Validator 5.3.6.Final
[2021-07-20 15:36:02,574][INFO][main][AnnotationConfigApplicationContext] - Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@276438c9: startup date [Tue Jul 20 15:36:02 HKT 2021]; root of context hierarchy
[2021-07-20 15:36:02,747][INFO][main][AutowiredAnnotationBeanPostProcessor] - JSR-330 'javax.inject.Inject' annotation found and supported for autowiring
[2021-07-20 15:36:03,352][INFO][main][SpringApplication] - 
______           _     _                _____
| ___ \         (_)   | |              /  ___|
| |_/ /___  __ _ _ ___| |_ _ __ _   _  \ `--.  ___ _ ____   _____ _ __
|    // _ \/ _` | / __| __| '__| | | |  `--. \/ _ \ '__\ \ / / _ \ '__|
| |\ \  __/ (_| | \__ \ |_| |  | |_| | /\__/ /  __/ |   \ V /  __/ |
\_| \_\___|\__, |_|___/\__|_|   \__, | \____/ \___|_|    \_/ \___|_|
            __/ |                __/ |
           |___/                |___/


[2021-07-20 15:36:03,470][INFO][main][AnnotationConfigApplicationContext] - Refreshing org.springframework.context.annotation.AnnotationConfigApplicationContext@159f197: startup date [Tue Jul 20 15:36:03 HKT 2021]; parent: org.springframework.context.annotation.AnnotationConfigApplicationContext@276438c9
[2021-07-20 15:36:05,213][INFO][main][AnnotationConfigApplicationContext] - Closing org.springframework.context.annotation.AnnotationConfigApplicationContext@159f197: startup date [Tue Jul 20 15:36:03 HKT 2021]; parent: org.springframework.context.annotation.AnnotationConfigApplicationContext@276438c9
[2021-07-20 15:36:05,217][INFO][Thread-2][AnnotationConfigApplicationContext] - Closing org.springframework.context.annotation.AnnotationConfigApplicationContext@276438c9: startup date [Tue Jul 20 15:36:02 HKT 2021]; root of context hierarchy
```

#### registry-integration-gc.log
```bash
[0.017s][info][gc] Using G1
[0.021s][info][gc,init] Version: 16.0.1+9-24 (release)
[0.021s][info][gc,init] CPUs: 8 total, 8 available
[0.021s][info][gc,init] Memory: 16384M
[0.021s][info][gc,init] Large Page Support: Disabled
[0.021s][info][gc,init] NUMA Support: Disabled
[0.021s][info][gc,init] Compressed Oops: Enabled (Zero based)
[0.021s][info][gc,init] Heap Region Size: 1M
[0.021s][info][gc,init] Heap Min Capacity: 512M
[0.021s][info][gc,init] Heap Initial Capacity: 512M
[0.021s][info][gc,init] Heap Max Capacity: 512M
[0.021s][info][gc,init] Pre-touch: Disabled
[0.021s][info][gc,init] Parallel Workers: 4
[0.021s][info][gc,init] Concurrent Workers: 1
[0.021s][info][gc,init] Concurrent Refinement Workers: 4
[0.021s][info][gc,init] Periodic GC: Disabled
[0.022s][info][gc,metaspace] CDS archive(s) mapped at: [0x0000000800000000-0x0000000800bb9000-0x0000000800bb9000), size 12292096, SharedBaseAddress: 0x0000000800000000, ArchiveRelocationMode: 0.
[0.022s][info][gc,metaspace] Compressed class space mapped at: 0x0000000800c00000-0x0000000840c00000, reserved size: 1073741824
[0.022s][info][gc,metaspace] Narrow klass base: 0x0000000800000000, Narrow klass shift: 3, Narrow klass range: 0x100000000
[5.257s][info][gc,start    ] GC(0) Pause Young (Normal) (G1 Evacuation Pause)
[5.258s][info][gc,task     ] GC(0) Using 4 workers of 4 for evacuation
[5.279s][info][gc,phases   ] GC(0)   Pre Evacuate Collection Set: 0.1ms
[5.279s][info][gc,phases   ] GC(0)   Merge Heap Roots: 0.3ms
[5.279s][info][gc,phases   ] GC(0)   Evacuate Collection Set: 19.6ms
[5.279s][info][gc,phases   ] GC(0)   Post Evacuate Collection Set: 0.9ms
[5.279s][info][gc,phases   ] GC(0)   Other: 0.8ms
[5.279s][info][gc,heap     ] GC(0) Eden regions: 256->0(249)
[5.279s][info][gc,heap     ] GC(0) Survivor regions: 0->7(32)
[5.279s][info][gc,heap     ] GC(0) Old regions: 0->0
[5.279s][info][gc,heap     ] GC(0) Archive regions: 2->2
[5.279s][info][gc,heap     ] GC(0) Humongous regions: 0->0
[5.279s][info][gc,metaspace] GC(0) Metaspace: 13240K(13440K)->13240K(13440K) NonClass: 11705K(11776K)->11705K(11776K) Class: 1534K(1664K)->1534K(1664K)
[5.279s][info][gc          ] GC(0) Pause Young (Normal) (G1 Evacuation Pause) 256M->7M(512M) 21.881ms
[5.279s][info][gc,cpu      ] GC(0) User=0.05s Sys=0.02s Real=0.02s
[5.977s][info][gc,heap,exit] Heap
[5.977s][info][gc,heap,exit]  garbage-first heap   total 524288K, used 38635K [0x00000007e0000000, 0x0000000800000000)
[5.977s][info][gc,heap,exit]   region size 1024K, 38 young (38912K), 7 survivors (7168K)
[5.977s][info][gc,heap,exit]  Metaspace       used 14544K, committed 14784K, reserved 1064960K
[5.977s][info][gc,heap,exit]   class space    used 1699K, committed 1792K, reserved 1048576K
```

#### registry-startup.log
```bash
[2021-07-20 15:36:02,497][INFO][main][RegistryApplication] - Starting RegistryApplication on wilson-mbp.lan with PID <PID> (/ProjectPath/Downloads/sofa-registry-integration/registry-integration/registry-integration.jar started by wilson in /ProjectPath/Downloads/sofa-registry-integration/registry-integration)
[2021-07-20 15:36:02,500][INFO][main][RegistryApplication] - No active profile set, falling back to default profiles: default
[2021-07-20 15:36:03,093][INFO][main][RegistryApplication] - Started RegistryApplication in 2.489 seconds (JVM running for 3.848)
[2021-07-20 15:36:03,360][INFO][main][RegistryApplication] - No active profile set, falling back to default profiles: default
```

Attached the logs files: [logs.zip](https://github.com/sofastack/sofa-registry/files/6846474/logs.zip)

It seems the JAVA 16.0.1 is not compatible with the sofa-registry. 
I also tested to build and run the sofa-registry under JAVA SE 8, it works fine and does not have any errors.

So, as a workaround, could we suggest others run the sofa-registry under JAVA SE 8 rather than JAVA OpenJDK 16.0.*?

--- a/README.md
+++ b/README.md
@@ -24,6 +24,8 @@ SOFARegistry 是蚂蚁金服开源的一个生产级、高时效、高可用的�
 
 运行需要 JDK 6 及以上，服务端运行需要 JDK 8及以上。
 
+****推荐使用JDK 8，JDK 16尚未被测试，可能会有兼容性问题。**
+
 ## 文档
 
 - [快速开始](https://www.sofastack.tech/sofa-registry/docs/Server-QuickStart)

